src/watch/watch-proxy.ts

```diff
@@ -1,25 +1,32 @@
-import type { RollupWatcher } from '../rollup/types';
+import { handleError } from '../../cli/logging';
+import type { RollupOptions, RollupWatcher } from '../rollup/types';
 import { ensureArray } from '../utils/ensureArray';
 import { error, errorInvalidOption } from '../utils/error';
-import type { GenericConfigObject } from '../utils/options/options';
+import { mergeOptions } from '../utils/options/mergeOptions';
 import { WatchEmitter } from './WatchEmitter';
 import { loadFsEvents } from './fsevents-importer';
 
-export default function watch(configs: GenericConfigObject[] | GenericConfigObject): RollupWatcher {
+export default function watch(configs: RollupOptions[] | RollupOptions): RollupWatcher {
 	const emitter = new WatchEmitter() as RollupWatcher;
-	const configArray = ensureArray(configs);
-	const watchConfigs = configArray.filter(config => config.watch !== false);
-	if (watchConfigs.length === 0) {
-		return error(
-			errorInvalidOption(
-				'watch',
-				'watch',
-				'there must be at least one config where "watch" is not set to "false"'
-			)
-		);
-	}
-	loadFsEvents()
-		.then(() => import('./watch'))
-		.then(({ Watcher }) => new Watcher(watchConfigs, emitter));
+
+	(async () => {
+		const optionsList = await Promise.all(ensureArray(configs).map(config => mergeOptions(config)));
+		const watchOptionsList = optionsList.filter(config => config.watch !== false);
+		if (watchOptionsList.length === 0) {
+			return error(
+				errorInvalidOption(
+					'watch',
+					'watch',
+					'there must be at least one config where "watch" is not set to "false"'
+				)
+			);
+		}
+		await loadFsEvents();
+		const { Watcher } = await import('./watch');
+		new Watcher(watchOptionsList, emitter);
+	})().catch(error => {
```

Comment:
I find IIFEs are always a little hard to read, maybe just extract another function?